src/horust/supervisor/process_spawner.rs

```diff
@@ -188,9 +201,15 @@ fn exec(
     unistd::setsid()?;
     // Set the user ID
     unistd::setuid(uid)?;
-    #[cfg(target_os = "linux")]
-    unistd::execvpe(program_name.as_ref(), arg_cptr.as_ref(), env_cptr.as_ref())?;
-    #[cfg(not(target_os = "linux"))]
-    unistd::execve(program_name.as_ref(), arg_cptr.as_ref(), env_cptr.as_ref())?;
+    let path = if program_name.contains("/") {
+        program_name.to_string()
+    } else {
+        find_program_path(&program_name)?
```

Comment:
the path should be passed to this function as argument, because find_program_path and program_name.to_string() will need the allocator. Check the safety documentation on this function. You can update the exec_args function. We can run both computation for linux and non linux using execve. I can't see any problems with it, can you?